ursula_cli/ursula_inventory.py

```diff
@@ -0,0 +1,195 @@
+        for host in self.get_hosts(ignore_limits_and_restrictions=True):
+            host.vars = combine_vars(host.vars,
+                                     self.get_host_variables(host.name))
+            self.get_host_vars(host)
+
+    def _get_defaults(self):
+        p, cfg_path = load_config_file()
+        defaults_file = get_config(p, DEFAULTS, 'var_defaults_file',
+                                   'ANSIBLE_VAR_DEFAULTS_FILE', None)
+        if not defaults_file:
+            return None
+
+        ursula_env = self.host_list
+        defaults_path = os.path.join(ursula_env, defaults_file)
+        if os.path.exists(defaults_path):
+            with open(defaults_path) as fh:
+                return yaml.safe_load(fh)
+        return None
+
+
+def init_logfile():
+    config = ConfigParser()
+    config.read('ansible.cfg')
+
+    try:
+        cfg_log = config.get('defaults', 'log_path')
+    except (NoOptionError, NoSectionError):
+        cfg_log = None
+
+    if cfg_log:
+        logfile = os.path.expanduser(cfg_log)
+    else:
+        logfile = 'ursula.log'
+
+    if not os.path.exists(logfile):
+        with open(logfile, 'a'):
+            os.utime(logfile, None)
+
+
+def _initialize_logger(level=logging.DEBUG, logfile=None):
+    init_logfile()
+    global LOG
+    LOG.setLevel(level)
+
+    handler = logging.StreamHandler()
+    formatter = logging.Formatter("%(levelname)s: %(message)s")
+    handler.setFormatter(formatter)
+    LOG.addHandler(handler)
+
+
+def run(args, extra_args):
+
+    env = os.environ.get('URSULA_ENV')
+    if not env:
+        raise Exception("Environment not provided")
+
+    if not os.path.exists(env):
+        raise Exception("Environment '%s' does not exist" % env)
+
+    loader = DataLoader()
+
+    variable_manager = VariableManager()
+
+    inventory = UrsulaInventory(loader, variable_manager, env)
+
+    # populate the dict to output
+    result = {}
+
+    # Define the groups and host memberships
+    for host in inventory.get_hosts():
+        for group in host.get_groups():
+            ghosts = result.setdefault(group.name, {'hosts': []})['hosts']
+            if host.name not in ghosts:
+                result[group.name]['hosts'].append(host.name)
+
+    # Define the group variables (takes defaults into account)
+    for group in inventory.groups:
+        if group != 'ungrouped':
+            groupobj = inventory.get_group(group)
+            result[group]['vars'] = groupobj.vars
+
+    # Define the host vars in _meta to avoid --host calls
+    result['_meta'] = {'hostvars': inventory._vars_per_host}
+
+    if args.list:
+        print(json.dumps(result))
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description='Inventory script for Ursula')
+    parser.add_argument('--list', help='List the inventory',
+                        action='store_true', default=True),
+    parser.add_argument('--host', help='Show a specific host detail')
+    return parser.parse_known_args()
```

Comment:
personally i'd probably just `parse_args` because I'd want to know if and when Ansible started passing non-standard arguments to this script, raising the error and exiting